comm.go

```diff
@@ -175,19 +175,15 @@ func (p *PubSub) handleSendingMessages(ctx context.Context, s network.Stream, ou
 
 	defer s.Close()
 	for {
```

Comment:
```suggestion
	for ctx.Err() == nil {
```

(existing bug, but the new queue implementation could exacerbate it).